### PR TITLE
github: Switch to yaml issue templates

### PR DESCRIPTION
**Pull request by willcl-ark**
The new YAML templates provide more flexibility and can be designed to extract more information from users when submitting issues, avoiding initial back-and-forth when reports do not include enough background information to begin with.

Key differences:

* YAML format
* Allows us to require responses to certain questions
* Not currently compatible with GitLab (.md only)

This does keep the "Blank Issue" option at the bottom.

Testing this must be done with the master branch of the repo, which is slightly annoying for this repo. I have therefore pushed this to my own fork so that you can see the new templates, along with how the output is rendered in newly-created issues:

[github.com/willcl-ark/bitcoin/issues](https://github.com/willcl-ark/bitcoin/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)

I did make some minor changes to some of the template wording, but this change could also be a good time to add/remove additional questions.

This seems like a net-positive for me, setting aside the issue that if we ever migrated away from GitHub these might have to be ported back to *.md (or something else), but that seems easy-enough that this change would be worth it.

Curious to know what others think of this, and whether they would suggest adding any other questions to any of the templates as part of this update?

### CODE REVIEW

**Comment by fanquake** [comment]
Concept ACK

File: .github/ISSUE_TEMPLATE/config.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+blank_issues_enabled: true
+contact_links:
+  - name: Bitcoin Core Developers
+    url: "https:bitcoincore.org"
```
Comment (flack) [inline .github/ISSUE_TEMPLATE/config.yaml:4]: Shouldn't this be `https://bitcoincore.org`?

Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/config.yaml:4]: Yep!

Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/config.yaml]: Fixed in 6d67ee23dbbe007a910e7cebb0d3ee4d1268ca15

Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/config.yaml]: Commit is now b18032cfa6ee590f231448e67e05c862750ea61f since I fixed the config.yaml not being detected (must be config.yml)

**Comment by fanquake** [comment]
What are the next steps here? Undraft?

**Comment by pinheadmz** [comment]
concept ACK

great work! I can't think of anything to add or change

**Comment by willcl-ark** [comment]
I suppose one other question is whether having more/mandatory questions will put people off submitting issues, although as mentioned in OP there is still the "blank issue" option available (in small writing) at the bottom of the issues page...

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      required: true
+  - type: input
+    id: machine-specs
+    attributes:
+      label: Machine specifications
+      description: |
+        What are the specifications of the host machine?
+        e.g. OS/CPU and disk type, network connectivity
+    validations:
+      required: true
```
Comment (flack) [inline .github/ISSUE_TEMPLATE/bug.yml:90]: If you want to reduce the number of required fields, this might be a good candidate to make optional, since I guess there are quite a few issues that are not related to hardware (also, OS in the description is a bit redundant, since this is already asked in the question above)

Comment (flack) [reply .github/ISSUE_TEMPLATE/bug.yml:90]: P.S.: Also, if you want the user to seriously describe CPU, disk, network and so on, textarea might be better than single line

**Comment by fanquake** [comment]
> I suppose one other question is whether having more/mandatory questions will put people off submitting issues,

If people can't be bothered to supply what we think is the minimum required information, then we will likely be unable to debug anything anyways. However, sometimes (fixable/informative) issues do come in the form of a randonly pasted debug log or similar. So if there is still an option for someone to dump something, I think this is ok.

@achow101 @glozow @MarcoFalke @hebasto any thoughts here?

**Comment by willcl-ark** [comment]
> So if there is still an option for someone to dump something, I think this is ok.

It would also be possible to add a new `BLANK` issue template so that the blank issue template gets a big green button like the others (rather than needing to click the small hyperlink at the bottom), but IMO this would slightly defeat the point of requiring more info.

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 2 above)
Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/bug.yml:90]: Updated in 5527f7490

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+blank_issues_enabled: true
+contact_links:
+  - name: Bitcoin Core Security Policy
+    url: https://github.com/bitcoin/bitcoin/blob/master/SECURITY.md
+    about: View security policy
```
Comment (achow101) [inline .github/ISSUE_TEMPLATE/config.yml:5]: There's already a big green button for the security policy, so I don't think we need to add a second one manually.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  - type: textarea
+    id: logs
+    attributes:
+      label: Relevant log output
+      description: |
+        Please copy and paste any relevant log output. This is normally the contents of a `debug.log` or `config.log` file.
+        This will be automatically formatted into code, so no need for backticks. If longer than 10,000 chars use a pastebin type site and just leave a link.
+      render: shell
```
Comment (achow101) [inline .github/ISSUE_TEMPLATE/bug.yml:53]: The docs say that users can upload files as long as `render` is not set. I think it would be better to let reporters upload their log files rather than requiring them to use a paste site.

**Review by achow101 — COMMENTED** [review]
Concept ACK

> It would also be possible to add a new `BLANK` issue template so that the blank issue template gets a big green button like the others (rather than needing to click the small hyperlink at the bottom), but IMO this would slightly defeat the point of requiring more info.

I think the link at the bottom is fine. The people who would use it can probably figure it out.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        * Actions taken
+    validations:
+      required: true
+  - type: textarea
+    id: logs
+    attributes:
+      label: Relevant log output
+      description: |
+        Please copy and paste any relevant log output. This is normally the contents of a `debug.log` or `config.log` file.
+        This will be automatically formatted into code, so no need for backticks. If longer than 10,000 chars use a pastebin type site and just leave a link.
```
Comment (kouloumos) [inline .github/ISSUE_TEMPLATE/bug.yml:52]: I think that a good addition here could be some of the usual hints posted in bug investigation comments such as https://github.com/bitcoin/bitcoin/issues/27134#issuecomment-1438098095, which are:
>   You can find the debug.log in your [data dir](https://github.com/bitcoin/bitcoin/blob/master/doc/files.md#data-directory-location).
Please be aware that the debug log might contain personally identifying information.

**Review by kouloumos — COMMENTED** [review]
Concept ACK

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 4 above)
Comment (kouloumos) [reply .github/ISSUE_TEMPLATE/bug.yml:53]: This could also help to avoid scenarios where thousands of lines of pasted logs make the issue harder to read.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      required: true
+  - type: textarea
+    id: machine-specs
+    attributes:
+      label: Machine specifications
+      description: |
+        What are the specifications of the host machine?
+        e.g. OS/CPU and disk type, network connectivity
+    validations:
+      required: true
```
Comment (glozow) [inline .github/ISSUE_TEMPLATE/bug.yml:90]: Agree this could be an optional "additional info about env or machine" as it could be irrelevant to the issue / confusing. (feel free to ignore as this is just me bikeshedding)

**Review by glozow — COMMENTED** [review]
Concept ACK, I think this is an improvement and have no major issues with it.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/config.yml:5]: This config option enables the `Bitcoin Core Security Policy` row with the `Open` button which links to our `SECURITY.md`. The `Report a security vulnerability` row, with the big green `Report a vulnerability` button seems to be a GitHub default (which I don't know how to disable).

My preference would be to keep both, as otherwise the link to `SECURITY.md` via the `Report a vulnerability` button is from a small right menu item:

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 5 above)
Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/bug.yml]: Added in 3fa1185dd

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 6 above)
Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/bug.yml]: I agree that this is nice-to-have rather than essential (for most bug reports) and so made it optional in 3fa1185dd

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 4 above)
Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/bug.yml]: I made this change in 3fa1185dd. Seems like files can be added to the textarea with drag & drop

**Comment by willcl-ark** [comment]
Thanks for the comments. I have incorporated the changes into 3fa1185dd which is also live on my fork.

**Comment by hebasto** [comment]
Concept ACK.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (achow101) [reply .github/ISSUE_TEMPLATE/config.yml:5]: It looks like it's different depending on your repo permissions.

For me, the big green button says "View Policy" and clicking on it goes to https://github.com/willcl-ark/bitcoin/security/policy. For a repo I own/admin/maintain, it shows the page similar to the one in your screenshot.

**Review by glozow — COMMENTED** [review]
ACK 3fa1185dda3b000b9c3956422fd2351e40969dec

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (willcl-ark) [reply .github/ISSUE_TEMPLATE/config.yml:5]: Hmmm, for me as a non admin this repo now looks like:

image

The green `View Policy` button takes me to a page which includes "Policy" and "Advisories", defaulting to showing the security policy `SECURITY.md`.

The row below that, "Bitcoin Core Security Policy" with the non-green `Open` button links me straight to `SECURITY.md`, which I agree is in theory somewhat redundant. If there is a preference to remove the second row, "Bitcoin Core Security Policy" direct link, then we could do that?
